dts/arm/st/l4/stm32l462.dtsi

```diff
@@ -11,7 +11,7 @@
 		compatible = "st,stm32l462", "st,stm32l4", "simple-bus";
 
 		aes: aes@50060000 {
-			compatible = "st,stm32-aes";
+			compatible = "st,stm32l4-aes";
```

Comment:
Note: Alternatively, you can do:
```suggestion
			compatible = "st,stm32-aes", "st,stm32l4-aes";
```
This way, you benefit from `DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes)` being defined and it can simplify things like:
```diff
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp)
#define DT_DRV_COMPAT st_stm32_cryp
#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes)
#define DT_DRV_COMPAT st_stm32_aes
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32l4_aes)    ===> No more required
+#define DT_DRV_COMPAT st_stm32l4_aes
```

Reply:
Consequently, I introduced USE_HAL_GENERIC and USE_HAL_L4 to differentiate the nature of the two HALs